### PR TITLE
Add block to restore sanitised backup to preprod

### PR DESCRIPTION
**Pull request by enformatique11**
**Context:** 

The block is created to restore the sanitised production back up to the pre-prod environment. 
I want someone to look at it before I rerun it. 

Please see the run [here] (https://github.com/DFE-Digital/apply-for-qualified-teacher-status/actions/runs/3904343647/jobs/6670058497)

### CODE REVIEW

**Comment by enformatique11** [comment]
[Restore pre-prod nightly](https://trello.com/c/vujChwhA/1271-restore-pre-prod-nightly)

File: .github/workflows/database-backup.yml [hunk 1]
```diff
@@ -115,3 +115,68 @@ jobs:
           name: backup_sanitised
           path: backup_sanitised.sql
           retention-days: 3
+
+    job2:
+      name: Restore backup to Preprod
+      runs-on: ubuntu-latest
+      needs: [job1]
+      services:
```
Comment (NickGraham101) [inline .github/workflows/database-backup.yml]: What is the service being used for?  It doesn't look like it's used in this job

Comment (saliceti) [inline .github/workflows/database-backup.yml:123]: We don't need the postgres service as it's only used for the sanitisation and it's already done

File: .github/workflows/database-backup.yml [hunk 2]
```diff
@@ -115,3 +115,68 @@ jobs:
...
+      name: preprod
+      steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: Azure/login@v1
+        with:
+          creds: ${{ secrets.azure_credentials }}
+
+      - name: Set environment variables Production
```
Comment (NickGraham101) [inline .github/workflows/database-backup.yml]: Name still refers to `Production` when variable being set are preprod

File: .github/workflows/database-backup.yml [hunk 3]
```diff
@@ -115,3 +115,68 @@ jobs:
...
+        run: |
+          tf_vars_file=terraform/workspace_variables/preprod.tfvars.json
+          echo "KEY_VAULT_NAME=$(jq -r '.key_vault_name' ${tf_vars_file})" >> $GITHUB_ENV
+          echo "PAAS_SPACE=$(jq -r '.paas_space' ${tf_vars_file})" >> $GITHUB_ENV
+
+      - uses: Azure/get-keyvault-secrets@v1
+        id: get_secrets
+        with:
+          keyvault: ${{ env.KEY_VAULT_NAME }}
+          secrets: "BACKUP-STORAGE-CONNECTION-STRING,PAAS-USER,PAAS-PASSWORD"
```
Comment (NickGraham101) [inline .github/workflows/database-backup.yml:145]: What is the `BACKUP-STORAGE-CONNECTION-STRING` secret used for in this job?

File: .github/workflows/database-backup.yml [hunk 4]
```diff
@@ -115,3 +115,68 @@ jobs:
           name: backup_sanitised
           path: backup_sanitised.sql
           retention-days: 3
+
+    job2:
```
Comment (saliceti) [inline .github/workflows/database-backup.yml:119]: Can we give more explicit job names? Like backup_production and restore_to_preprod

File: .github/workflows/database-backup.yml [hunk 5]
```diff
@@ -115,3 +115,68 @@ jobs:
...
+          echo "KEY_VAULT_NAME=$(jq -r '.key_vault_name' ${tf_vars_file})" >> $GITHUB_ENV
+          echo "PAAS_SPACE=$(jq -r '.paas_space' ${tf_vars_file})" >> $GITHUB_ENV
+
+      - uses: Azure/get-keyvault-secrets@v1
+        id: get_secrets
+        with:
+          keyvault: ${{ env.KEY_VAULT_NAME }}
+          secrets: "BACKUP-STORAGE-CONNECTION-STRING,PAAS-USER,PAAS-PASSWORD"
+
+      - uses: DfE-Digital/keyvault-yaml-secret@v1
```
Comment (NickGraham101) [inline .github/workflows/database-backup.yml:147]: What is the `SLACK_WEBHOOK` used for in this job?

File: .github/workflows/database-backup.yml (hunk 4 above)
Comment (NickGraham101) [inline .github/workflows/database-backup.yml:119]: job name needs to be more descriptive, something like `restore-to-preprod`

File: .github/workflows/database-backup.yml [hunk 6]
```diff
@@ -115,3 +115,68 @@ jobs:
...
+          CF_USERNAME: ${{ steps.get_secrets.outputs.PAAS-USER }}
+          CF_PASSWORD: ${{ steps.get_secrets.outputs.PAAS-PASSWORD }}
+          CF_SPACE_NAME: ${{ env.PAAS_SPACE }}
+          INSTALL_CONDUIT: true
+
+      - name: Download Sanitised Backup
+        uses: actions/download-artifact@v3
+        with:
+          name: backup_sanitised
+          path: backup_sanitised.sql
```
Comment (NickGraham101) [inline .github/workflows/database-backup.yml:168]: Doesn't look like this input is needed from what the [docs](https://github.com/actions/download-artifact) say about it

File: .github/workflows/database-backup.yml [hunk 7]
```diff
@@ -115,3 +115,68 @@ jobs:
...
+          
+      - name: Restore sanitised backup to preprod
```
Comment (NickGraham101) [inline .github/workflows/database-backup.yml:170]: I'd comment this step out until you're happy the rest of the workflow works as expected

**Review by NickGraham101 — DISMISSED** [review]

File: .github/workflows/database-backup.yml (hunk 5 above)
Comment (enformatique11) [reply .github/workflows/database-backup.yml:147]: we dont need it

Comment (enformatique11) [reply .github/workflows/database-backup.yml:147]: removed

File: .github/workflows/database-backup.yml (hunk 6 above)
Comment (enformatique11) [reply .github/workflows/database-backup.yml:168]: I dont get this comment

File: .github/workflows/database-backup.yml (hunk 3 above)
Comment (enformatique11) [reply .github/workflows/database-backup.yml:145]: Don't require it --- removed

**Review by saliceti — APPROVED** [review]

**Pull request merged**